Feature/mc 14135 password policy (#320)
cloudops/cloudmc-api-docs: source/includes/administration/_organizations.md

`organization`<br/>*[Organization](#administration-organizations)* | The organization to which the verified domain belongs. *includes*:`id`,`name`, `entryPoint`.
`autoCreationEnabled`<br/>*boolean* | A boolean specifying whether to enable automatic end-user account creation upon successful OIDC login.
`verifiedDomains`<br/>*Array[[verified domains](#administration-get-verified-domains)]*| A list of verified domains (with VERIFIED status) for which successful matching OIDC logins will create new users.
`passwordPolicy`<br/>*object*  | The password policy assigned to the organization. 

> `passwordPolicy.constraints`<br/>*Array[Object]* | List of password policy constraints. Each constraint contains -

Reviewer: ```suggestion
`passwordPolicy.constraints`<br/>*Array[Object]* | List of password policy constraints.
```

As now we are documenting the fields using absolute paths so we can drop this.

> `passwordPolicy.constraints`<br/>*Array[]* | List of password policy constraints objects with the following fields.

Reviewer: Not sure but how about something like this?
```suggestion
`passwordPolicy.constraints`<br/>*Array[]* | List of password policy constraints. Each constraint contains -
- `name`: A string that represents the constraint name.
- `value`: An integer that represents the minimum value for the constraint.
- `isMandatory`: A boolean flag to indicate if the constraint is mandatory or not.
```

Author: @y0geshchaudhary Makes sense 👍🏻
Thank you for the suggestion!

Returns an HTTP status code 200, with an empty response body.

<!-------------------- GET ORGANIZATION PASSWORD POLICY -------------------->

### Get password policy
`GET /organizations/organization_id/password_policy`

Retrieve the password policy for the organization.

```shell

> # Retrieve the organization's security settings

Reviewer: ```suggestion
# Retrieve the organization's own or inherited password policy.
```

   -H "MC-Api-Key: your_api_key"
```
> The above command returns a JSON structured like this:

```json
{
    "data": [
        {
            "name": "min_password_length",
            "value": 8,
            "isMandatory": true
        },
        {
            "name": "min_lowercase_letters",
            "value": 1,
            "isMandatory": true
        },
        {
            "name": "min_uppercase_letters",
            "value": 1,
            "isMandatory": true
        },
        {
            "name": "min_numbers",
            "value": 1,
            "isMandatory": true
        },
        {
            "name": "min_special_characters",
            "value": 1,
            "isMandatory": true
        }
    ]
}
```
Attributes | &nbsp;
---- | -----------
*Array[Object]*| A list of password policy constraint objects with the following fields.
`name`<br/>*string* | The name of the constraint.
`value`<br/>*int* | The minimum value for the constraint.
`isMandatory`<br/>*boolean* | Flag to indicate if the constraint is mandatory or not.

<!-------------------- CREATE/UPDATE ORGANIZATION PASSWORD POLICY -------------------->

> ### Create/Update password policy

Reviewer: Should we update the `Update security settings` section instead of creating a new section for password policy as it's the same endpoint?

Author: IMO, since `password policy` is totally optional during the create/update of `security settings`, keeping this `PUT` call lightweight would be idealistic. One use-case is when the user is interested in 
1. Only Creating/updating the security settings  --> document that `password policy` fields are optional
2. Only Creating/updating the password policy --> document that `security settings ` fields are optional

The API doc here would just become too conditional IMO, and segregating them (even though they belong to just one endpoint) might simplify it with the desired use case. 

But yes, if we want to document these two functionalities in one reference, I am cool with that as well!
Let me know!

Reviewer: `if we want to document these two functionalities in one reference` - It would make sense to merge them in one because here we are documenting the APIs and not functionalities.

Similar to `autoCreationEnabled` and `verifiedDomains`, the password policy fields should go in the optional section.

Reviewer: We should also add an `info` section something like - `If there is no default role provided in the request then Guest will be saved as default role.`

`passwordPolicy.constraints`<br/>*Array[]* | List of password policy constraints objects with the following fields.
`name`<br/>*string* | The name of the constraint.
`value`<br/>*int* | The minimum value for the constraint.
`isMandatory`<br/>*boolean* | Flag to indicate if the constraint is mandatory or not.

Optional | &nbsp;
---- | ----
`defaultRole`<br/>*object*  | The role that will be assigned to new users logging into the organization with an email domain matching that of the organization's security settings. 
`autoCreationEnabled`<br/>*boolean* | A boolean specifying whether to enable automatic end-user account creation upon successful OIDC login.
`verifiedDomains`<br/>*Array[[verified domains](#administration-get-verified-domains)]*| A list of objects containing the ids of verified domains (with VERIFIED status) for which successful matching OIDC logins will create new users.

Returns an HTTP status code 200, with an empty response body.

<!-------------------- DELETE PASSWORD POLICY FOR ORGANIZATION -------------------->
### Delete password policy
`DELETE /organizations/:id/password_policy`

> Delete a password policy for an organization. Root reseller organization will not be able to delete its password policy. A sub-organization can delete its password policy

Reviewer: ```suggestion
Delete a password policy for an organization. Root reseller organization will not be able to delete its password policy. A reseller sub-organization can delete its password policy
```

`passwordPolicy`<br/>*object*  | The password policy that will be assigned to the organization. 
`passwordPolicy.constraints`<br/>*Array[Object]* | List of password policy constraints objects with the following fields.
`passwordPolicy.constraints.name`<br/>*string* | A string that represents the constraint name.
`passwordPolicy.constraints.value`<br/>*int* | An integer that represents the minimum value for the constraint.
`passwordPolicy.constraints.isMandatory`<br/>*boolean* | A boolean flag to indicate if the constraint is mandatory or not.

Returns an HTTP status code 200, with an empty response body.

- If `defaultRole` is not passed in the request:
   - The system will assign `Guest`as a default role to the organization.
- If `passwordPolicy` is passed in the request:
   - The system will create a new or update the existing password policy for the organization.
- If `autoCreationEnabled` and/or `verifiedDomains` is passed in the request:
   - The system will create a new or update the existing security settings for the organization.

> 

Reviewer: Looks good.
We can even put this info in a notice section something like -

![image](https://user-images.githubusercontent.com/5191515/111373028-d0f3d800-8671-11eb-8148-e4f7c0a8b8b7.png)

by putting it all inside `<aside class="notice"></aside>` tag.

Author: I tried using the `<aside class="notice"></aside>` tag, but the other formatting options go for a toss in that case. IMO, the previous one with proper formatting also looks neet.

![image](https://user-images.githubusercontent.com/20757399/111376139-d6efb600-867d-11eb-8d55-6e220ce56d8a.png)

Reviewer: Thanks for checking.

```
> The above command returns a JSON structured like this:

```json
{
    "data": [
        {
            "name": "min_password_length",
            "value": 8,
            "isMandatory": true
        },
        {
            "name": "min_lowercase_letters",
            "value": 1,
            "isMandatory": true
        },
        {
            "name": "min_uppercase_letters",
            "value": 1,
            "isMandatory": true
        },
        {
            "name": "min_numbers",
            "value": 1,
            "isMandatory": true
        },
        {
            "name": "min_special_characters",
            "value": 1,
            "isMandatory": true
        }
    ]
}
```
Attributes | &nbsp;
---- | -----------
*Array[Object]*| A list of password policy constraint objects with the following fields.
`name`<br/>*string* | The name of the constraint.
`value`<br/>*int* | The minimum value for the constraint.
`isMandatory`<br/>*boolean* | Flag to indicate if the constraint is mandatory or not.

<!-------------------- DELETE PASSWORD POLICY FOR ORGANIZATION -------------------->
### Delete password policy
`DELETE /organizations/:id/password_policy`

> Delete a password policy for an organization. Root reseller organization will not be able to delete its password policy. A reseller sub-organization can delete its password policy

Reviewer: Small nitpick - a missing period in the end.
```suggestion
Delete a password policy for an organization. Root reseller organization will not be able to delete its password policy. A reseller sub-organization can delete its password policy.
```
